### PR TITLE
Fix replace handling

### PR DESCRIPTION
**Pull request by Lucus16**
This PR builds on https://github.com/adisbladis/vgo2nix/pull/27

I added a test for URL replacements based on how they are used elsewhere on github. Note that this URL replacement isn't useful, the backing git repositories are still the same, but it works fine as a test and is unlikely to break. Most real world examples involve pull requests that will eventually be deleted.

I then wrote URL replacement support which passes the tests. It's not a very elegant solution but it seems to work.

### CODE REVIEW

**Comment by matt-snider** [comment]
I'm currently trying to package [ProtonMail/proton-bridge](https://github.com/ProtonMail/proton-bridge) but was stumped when I ran into #22.

I tried your fork and it worked, so first of all thanks for sharing your solution! :+1:

I did have one issue, which was that I had to modify [go.mod](https://github.com/ProtonMail/proton-bridge/blob/b91c286332386d004d004714f6ede2e68870b0d0/go.mod#L8) since the version and commit for the `docker-credential-helpers` and `bcrypt` appear to be placeholders and I got an error about invalid commits. I'm not familiar enough with Go to say if this is an issue with vgo2nix or whether the `go.mod` is doing something non-standard.

File: tests/test_replace_url/go.mod [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module testmodule
+
+go 1.11
+
+require golang.org/x/text v0.3.2
```
Comment (Lucus16) [inline tests/test_replace_url/go.mod:5]: This should fix the test to check that the replaced version does not get used.

```suggestion
require golang.org/x/text v0.0.0-00010101000000-000000000000
```

**Comment by Lucus16** [comment]
It is normal for `go.mod` to contain invalid commits when a `replace` with a valid commit is given. My code incorrectly makes use of the replaced version to retrieve one of the paths.

**Comment by matt-snider** [comment]
@Lucus16 Yeah it seems like I spoke to soon. The deps.nix file is now generated (before it error'd out), but looking at it closer I see that the original dependencies are still listed there. I guess that's what you mean in your last sentence.

**Comment by SuperSandro2000** [comment]
This PR fixes some of the bugs master has but packages like ``dmitri.shuralyov.com/gpu/mtl`` are still not working.

**Pull request closed without merging**